examples/martingale.py

```diff
@@ -0,0 +1,183 @@
+import alpaca_trade_api as tradeapi
+
+# Utility to truncate a float value to a certain number of decimal places.
+# We'll use this to see if a "penny level" was crossed when we compare prices.
+# This is necessary because a price can change by 1/100th of a penny, but we
+# can only trade at full-penny increments.
+def truncate(val, decimal_places):
+    return int(val * 10**decimal_places) / 10**decimal_places
+
+
+# The MartingaleTrader bets that streaks of increases or decreases in a stock's
+# price are likely to break, and increases its bet each time it is wrong.
+class MartingaleTrader(object):
+    def __init__(self):
+        self.key_id = "REPLACE_ME"
+        self.secret_key = "REPLACE_ME"
+        self.base_url = 'https://paper-api.alpaca.markets'
+
+        # The symbol we will be trading
+        self.symbol = 'SPY'
+
+        # How many seconds we will wait in between updating the streak values
+        self.tick_size = 5
+        self.tick_index = 0
+
+        # The percentage of our buying power that we will allocate to a new
+        # position after a streak reset
+        self.base_bet = 10
+
+        # These variables track the information about the current streak
+        self.streak_count = 0
+        self.streak_start = 0
+        self.streak_increasing = True
+
+        # When this variable is not None, we have an order open
+        self.current_order = None
+
+        # The closing price of the last aggregate we saw
+        self.last_price = 0
+
+        # The connection to the Alpaca API
+        self.api = tradeapi.REST(
+            self.key_id,
+            self.secret_key,
+            self.base_url
+        )
+
+        # Cancel any open orders so they don't interfere with this script
+        self.api.cancel_all_orders()
+
+        try:
+            self.position = int(self.api.get_position(self.symbol).qty)
+        except:
+            # No position exists
+            self.position = 0
+
+        # Figure out how much money we have to work with, accounting for margin
+        account_info = self.api.get_account()
+        self.equity = float(account_info.equity)
+        self.margin_multiplier = float(account_info.multiplier)
+        total_buying_power = self.margin_multiplier * self.equity
+        print(f'Initial total buying power = {total_buying_power}')
+
+
+    def start_trading(self):
+        conn = tradeapi.StreamConn(
```

Comment:
Isnt trade_updates unauthorized with paper trading api keys?

Reply:
No - you might be getting it confused with the issue where paper keys are incorrectly accepted by the live API.

Reply:
Ah that might be it